tensornetwork/tensor.py

```diff
@@ -0,0 +1,201 @@
+    wheras ravel returns a view when possible.
+    """
+    size = self.size
+    flat = self.reshape([size,]).copy()
+    return flat
+
+  def fill(self, val: float):
+    """ Returns a `Tensor` filled with the scalar value `val`.
+    Args:
+      val: The scalar value.
+    Returns:
+      filled: A `Tensor` of the same shape as this one whose entries are all
+              `val`.
+    """
+    vals = np.zeros(self.shape)
+    vals = val
+    mask = np.ones(self.shape)
+    filled_arr = self.backend.index_update(self.array, mask, vals)
+    return Tensor(filled_arr, backend=self.backend)
+
+  def hconj(self, perm: Optional[Sequence[int]] = None) -> "Tensor":
+    """ The Hermitian conjugated tensor; e.g. the complex conjugate tranposed
+    by the permutation set be `axes`. By default the axes are reversed.
+    Args:
+      perm: The permutation. If None (default) the index order is reversed.  
+    Returns:
+      The Hermitian conjugated `Tensor`.
+    """
+    dag = self.conj().transpose(perm=perm)
+    return dag
+
+  def ravel(self):
+    """Return a new `Tensor` with the same number of elements but only one
+    dimension.
+    """
+    size = self.size
+    flat = self.reshape(shape=[size,])
+    return flat
+
+  def reshape(self, shape: Sequence[int]) -> "Tensor":
+    """Return a new `Tensor` with the same data but a new shape `shape`.
+    Args:
+      shape: The new shape.
+    Returns:
+      The reshaped `Tensor`.
+    """
+    reshaped = self.backend.reshape(self.array, shape)
+    return Tensor(reshaped, backend=self.backend)
+
+  def squeeze(self):
+    """Return a new `Tensor` with all axes of size 1 eliminated.
+    """
+    shape = self.shape
+    squeezed_shape = [d for d in shape if d != 1]
+    return self.reshape(squeezed_shape)
+
+  def transpose(self, perm: Optional[Sequence[int]] = None) -> "Tensor":
+    """ Return a new `Tensor` transposed according to the permutation set
+    by `axes`. By default the axes are reversed.
+    Args:
+      axes: The permutation. If None (default) the index order is reversed.
+    Returns:
+      The transposed `Tensor`.
+    """
+    array_T = self.backend.transpose(self.array, perm=perm)
+    return Tensor(array_T, backend=self.backend)
+
+  def __mul__(self, other: Union["Tensor", float]) -> "Tensor":
+    if isinstance(other, Tensor):
+      if self.backend.name != other.backend.name:
+        errstr = (f"Backends {self.backend.name} and {other.backend.name} did"
+                  f"not agree.")
+        raise ValueError(errstr)
+      other = other.array
+    array = self.backend.multiply(self.array, other)
+    return Tensor(array, backend=self.backend)
+
+  __rmul__ = __mul__
+
+  def __truediv__(self, other: Union["Tensor", float]) -> "Tensor":
+    if isinstance(other, Tensor):
+      if self.backend.name != other.backend.name:
+        errstr = (f"Backends {self.backend.name} and {other.backend.name} did"
+                  f"not agree.")
```

Comment:
Change to `f"Given backends are inconsistent. Found '{self.backend.name}' and '{other.backend.name}'"`